Userland/Services/SpiceAgent/Message.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2023, Caoimhe Byrne <caoimhebyrne06@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Forward.h>
+#include <AK/Vector.h>
+
+namespace SpiceAgent {
+
+static constexpr u32 AGENT_PROTOCOL = 1;
+
+// Used to communicate what the client/or server is capable of
+// Not a lot of documentation is available for all of these, but the headers contain some information
+// https://gitlab.freedesktop.org/spice/spice-protocol/-/blob/master/spice/vd_agent.h
+enum class Capability : u32 {
+    MouseState = 0,
+    MonitorsConfig,
+    Reply,
+    Clipboard,
+    DisplayConfig,
+    ClipboardByDemand,
+    ClipboardSelection,
+    SparseMonitorsConfig,
+    GuestLineEndLF,
+    GuestLineEndCRLF,
+    MaxClipboard,
+    AudioVolumeSync,
+    MonitorsConfigPosition,
+    FileTransferDisabled,
+    FileTransferDetailedErrors,
+    GraphicsCardInfo,
+    ClipboardNoReleaseOnRegrab,
+    ClipboardGrabSerial
+};
+
+class Message {
+public:
+    // The spice protocol headers contain a bit of documentation about these, but nothing major:
+    // https://gitlab.freedesktop.org/spice/spice-protocol/-/blob/master/spice/vd_agent.h
+    enum class Type : u32 {
+        MouseState = 1,
+        MonitorsConfig,
+        Reply,
+        Clipboard,
+        DisplayConfig,
+        AnnounceCapabilities,
+        ClipboardGrab,
+        ClipboardRequest,
+        ClipboardRelease,
+        FileTransferStart,
+        FileTransferStatus,
+        FileTransferData,
+        Disconnected,
+        MaxClipboard,
+        VolumeSync,
+        GraphicsDeviceInfo
+    };
+
+    Message(Type type)
+        : m_type(type)
+    {
+    }
+
+    Type type() { return m_type; }
+
+    // Used for sending messages to the spice server
+    virtual ErrorOr<void> write_to_stream(AK::Stream& stream);
+
+    // Used for debugging
+    virtual ErrorOr<String> debug_description();
+
+    virtual ~Message() = default;
+
+private:
+    Type m_type { Message::Type::MouseState };
```

Comment:
Not sure how I feel about the choice of default value here, but I'm also not sure what else to pick or whether we can just completely skip initializing it since we have to pass it to the constructor anyways.

Reply:
I don't see why we need to initialize it here